gh-badges/lib/make-badge.js

```diff
@@ -1,104 +1,21 @@
 'use strict'
 
-const fs = require('fs')
-const path = require('path')
-const SVGO = require('svgo')
-const dot = require('dot')
-const anafanafo = require('anafanafo')
+const camelcase = require('camelcase')
 const { normalizeColor, toSvgColor } = require('./color')
-
-// cache templates.
-const templates = {}
-const templateFiles = fs.readdirSync(path.join(__dirname, '..', 'templates'))
-dot.templateSettings.strip = false // Do not strip whitespace.
-templateFiles.forEach(async filename => {
-  if (filename[0] === '.') {
-    return
-  }
-  const templateData = fs
-    .readFileSync(path.join(__dirname, '..', 'templates', filename))
-    .toString()
-  const extension = path.extname(filename).slice(1)
-  const style = filename.slice(0, -`-template.${extension}`.length)
-  // Compile the template. Necessary to always have a working template.
-  templates[style] = dot.template(templateData)
-  // Substitute dot code.
-  const mapping = new Map()
-  let mappingIndex = 1
-  const untemplatedSvg = templateData.replace(/{{.*?}}/g, match => {
-    // Weird substitution that currently works for all templates.
-    const mapKey = `99999990${mappingIndex}.1`
-    mappingIndex++
-    mapping.set(mapKey, match)
-    return mapKey
-  })
-
-  const svgo = new SVGO()
-  const { data, error } = await svgo.optimize(untemplatedSvg)
-
-  if (error !== undefined) {
-    console.error(
-      `Template ${filename}: ${error}\n` +
-        '  Generated untemplated SVG:\n' +
-        `---\n${untemplatedSvg}---\n`
-    )
-    return
-  }
-
-  // Substitute dot code back.
-  let svg = data
-  const unmappedKeys = []
-  mapping.forEach((value, key) => {
-    let keySubstituted = false
-    svg = svg.replace(RegExp(key, 'g'), () => {
-      keySubstituted = true
-      return value
-    })
-    if (!keySubstituted) {
-      unmappedKeys.push(key)
-    }
-  })
-  if (unmappedKeys.length > 0) {
-    console.error(
-      `Template ${filename} has unmapped keys ` +
-        `${unmappedKeys.join(', ')}.\n` +
-        '  Generated untemplated SVG:\n' +
-        `---\n${untemplatedSvg}\n---\n` +
-        '  Generated template:\n' +
-        `---\n${svg}\n---\n`
-    )
-    return
-  }
-
-  templates[style] = dot.template(svg)
-})
-
-function escapeXml(s) {
-  if (s === undefined || typeof s !== 'string') {
-    return undefined
-  } else {
-    return s
-      .replace(/&/g, '&amp;')
-      .replace(/</g, '&lt;')
-      .replace(/>/g, '&gt;')
-      .replace(/"/g, '&quot;')
-      .replace(/'/g, '&apos;')
-  }
-}
-
-function capitalize(s) {
-  return s.charAt(0).toUpperCase() + s.slice(1)
-}
+const badgeRenderers = require('./badge-renderers')
 
 /*
 note: makeBadge() is fairly thinly wrapped so if we are making changes here
 it is likely this will impact on the package's public interface in index.js
 */
 module.exports = function makeBadge({
   format,
-  template,
+  template = 'flat',
   text,
+  colorscheme,
   color,
+  colorA,
```

Comment:
At first glance I am surprised to see `colorscheme`, `colorA`, and `colorB` added here since they were deprecated and (I presumed) were potentially going to be dropped entirely as part of 3.x

Reply:
Good catch :+1: #3637 was originally targetted at `master`. When I rebased that stuff onto `v3-dev`, I hadn't realised I was accidentally undoing some of the work from https://github.com/badges/shields/pull/3881 I'll fix this.